### ISSUE TITLE
Add commit message lints

### ISSUE THREAD

**Issue by kchibisov** [rust-windowing/winit#3545]
Unfortunatelly, it's impossible for everyone to at least wrap their commit messages. After numerous warnings folks can't still do such things.

Thus, the squash merge will be disabled, and the only option will be rebase with the commit lint enabled for each commit in the series.

**Comment by madsmtm** [comment]
I'm probably one of the offenders here, but disagree that this is a priority, IMO just get your editor / terminal / git to wrap the lines for you if you want them wrapped.

That said, I'm fine with CI jobs or something to fix this, but I believe there is great value in _not_ telling contributors to rebase, and squashing instead.

**Comment by kchibisov** [comment]
>That said, I'm fine with CI jobs or something to fix this, but I believe there is great value in not telling contributors to rebase, and squashing instead.

You can do that yourself if you want to, I can't be bothered to deal with these commit messages anymore.

I told about that numerous times during the year(I think), so no it's a priority, since mostly all commit messages expept for the ones from @notgull and me are formatted like garbage.

**Comment by daxpedda** [comment]
I agree that this is not a priority. It would also be good to agree on something and write it down somewhere. Personally I couldn't care less about wrapping commit messages (to a degree ofc), but if we agree on something and write it down somewhere I'm happy to oblige.

I don't like squashing either, fast-forward-merging anybody?

**Comment by kchibisov** [comment]
I mean there's nothing to agree, it's a standard to wrap body at 72 and title around 50 chars. I also asked to strip garbage github adds, because I can't be bothered to read any of that and it just leaves a feeling that no one cares to write anything useful in the end of the day.

![image](https://github.com/rust-windowing/winit/assets/27620401/5680aba7-9e55-40ba-ab6b-04b9e1645958)


Or like that 

 
![image](https://github.com/rust-windowing/winit/assets/27620401/11ba1ff7-5426-4ca6-a004-5118c855e688)

**Comment by kchibisov** [comment]
>All I see is an issue with your terminal window being too wide / git not automatically wrapping the message to a readable width like any normal text-box in a modern system otherwise does ;).

Ofc, just read yours and then e.g. John's or mine. Or read the alacritty's messages. It's feels like you throw garbage which no-one cares about and then dance around the tree because your thing got merged.

And no, it's not an issue with my system in particular, you should just follow what git uses by default and at least put a bit of effort into writing about the work you've done.

**Comment by daxpedda** [comment]
> It's feels like you throw garbage which no-one cares about and then dance around the tree because your thing got merged.

I don't believe this language is appropriate. Please remain on topic.

---

> what git uses by default

I'm not aware of git imposing any rules by default? What are you referring to here?

**Comment by kchibisov** [comment]
>I don't believe this language is appropriate. Please remain on topic.

I was polite about this during the year. Mostly all my suggestions were ignored and they don't want to improve on purpose resulting in lower code quality in the project, because git commit messages are part of that. Given that I can't stand it anymore I take it personally and treat as a threat it as power abuse since 99% of the time I rewrite their commit messages for them,
which they still not appreciate and continue to merge arbitrary garbage. If they can't behave, I'll make the rules stricter for everyone, including good participants.

>I'm not aware of git imposing any rules by default? What are you referring to here?

50 and 72 wrapping is de-facto wrapping in git.

**Comment by kchibisov** [comment]
I've disabled squash and forced explicit review on each PR.

**Comment by madsmtm** [comment]
I've set up a Git `commit-msg` hook now.

**Comment by kchibisov** [comment]
>I've set up a Git commit-msg hook now.

For yourself or what do you mean by that?

**Comment by madsmtm** [comment]
I'll try to set up a CI job that checks this, so that we can re-enable squash merging

**Comment by kchibisov** [comment]
The issue is that CI won't be able to prevent squash merging, because you write commit messages on github, thus CI won't run for it.

The only way CI for commits can work is for rebase itself. The squash will be re-enabled eventually.

**Comment by daxpedda** [comment]
I think we should agree on some rules in https://github.com/rust-windowing/winit/pull/3549 first.

**Comment by kchibisov** [comment]
I think the point is to lint on width at least. The lints for commit messages are present in e.g. systemd, etc, iirc.
